SimpleWAWS/Code/CsmExtensions/CsmResourceGroupExtensions.cs

```diff
@@ -300,23 +300,60 @@ public static partial class CsmManager
+        {
+            var site = new Site(resourceGroup.SubscriptionId, resourceGroup.ResourceGroupName, nameGenerator(), siteKind);
+            site.Kind = siteKind;
+            await resourceGroup.LoadServerFarms(null);
+            ServerFarm serverFarm = new ServerFarm(resourceGroup.SubscriptionId,resourceGroup.ResourceGroupName,Constants.DefaultServerFarmName,
+                resourceGroup.GeoRegion);
+                var csmSiteResponse =
+                    await
+                        csmClient.HttpInvoke(HttpMethod.Put, ArmUriTemplates.Site.Bind(site),
+                            new
+                            {
+                                properties =
+                                    new { serverFarm = serverFarm,
+                                        sku = Constants.TryAppServiceSku},
+                                location = resourceGroup.GeoRegion,
+                                kind = siteKind
+                            });
+                await csmSiteResponse.EnsureSuccessStatusCodeWithFullError();
+                var csmSite = await csmSiteResponse.Content.ReadAsAsync<CsmWrapper<CsmSite>>();
+
+                return await Load(site, csmSite);
+        }
+        private static async Task<Site> CreateSiteUsingLatestVersion(ResourceGroup resourceGroup, Func<string> nameGenerator, string siteKind = null)
         {
-            var site = new Site(resourceGroup.SubscriptionId, resourceGroup.ResourceGroupName, nameGenerator());
-            var csmSiteResponse = await csmClient.HttpInvoke(HttpMethod.Put, ArmUriTemplates.Site.Bind(site), new { properties = new { }, location = resourceGroup.GeoRegion });
+            var site = new Site(resourceGroup.SubscriptionId, resourceGroup.ResourceGroupName, nameGenerator(), siteKind);
+            await resourceGroup.LoadServerFarms(null);
+            ServerFarm serverFarm = new ServerFarm(resourceGroup.SubscriptionId, resourceGroup.ResourceGroupName, Constants.DefaultServerFarmName,
+                resourceGroup.GeoRegion);
+            var csmSiteResponse =
+                await
+                    csmClient.HttpInvoke(HttpMethod.Put, ArmUriTemplates.LatestSiteApiVersionTemplate.Bind(site),
+                        new
+                        {
+                            properties =
+                                new
+                                {
+                                    serverFarmId=serverFarm.CsmId,
+                                    sku = Constants.TryAppServiceSku
+                                },
+                            location = resourceGroup.GeoRegion,
+                            kind = siteKind
+                        });
             await csmSiteResponse.EnsureSuccessStatusCodeWithFullError();
-
             var csmSite = await csmSiteResponse.Content.ReadAsAsync<CsmWrapper<CsmSite>>();
 
             return await Load(site, csmSite);
         }
```

Comment:
`CreateSite` and `CreateSiteUsingLatestVersion` have a lot of copy paste code. can you refactor them?